### PR TITLE
Fix bug of  automount-shares script

### PR DESCRIPTION
**Pull request by kofj**
fixed to support unicode file name.

### CODE REVIEW

File: rootfs/rootfs/etc/rc.d/automount-shares [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -e
 # VirtualBox Guest Additions
 # - this will bail quickly and gracefully if we're not in VBox
 if modprobe vboxguest &> /dev/null && modprobe vboxsf &> /dev/null; then
-	mountOptions='defaults'
+	mountOptions='iocharset=utf8'
```
Comment (tianon) [inline rootfs/rootfs/etc/rc.d/automount-shares]: According to the [`mount.vboxsf` man page](http://manpages.ubuntu.com/manpages/trusty/man8/mount.vboxsf.8.html), `utf8` is the default `iocharset`, so I'm a little confused why this would be necessary.

**Comment by SvenDowideit** [comment]
mmm, I recon LGTM - II recal reading at least 2 reports that utf8 Usernames on Windows failed...

but I only found https://github.com/boot2docker/boot2docker/issues/687 which is a different, though related issue.

File: rootfs/rootfs/etc/rc.d/automount-shares (hunk 1 above)
Comment (kofj) [inline rootfs/rootfs/etc/rc.d/automount-shares]: @tianon I have read the usage info,and tested the default iocharset.But,it didn't work,so I added it.

**Comment by SvenDowideit** [comment]
ping @tianon

File: rootfs/rootfs/etc/rc.d/automount-shares [hunk 2]
```diff
@@ -18,7 +18,7 @@ if modprobe vboxguest &> /dev/null && modprobe vboxsf &> /dev/null; then
 		name="${2:-$dir}"
 		
 		mkdir -p "$dir" 2>/dev/null
-		if ! mount -t vboxsf -o "$mountOptions" "$name" "$dir" 2>/dev/null; then
+		if ! mount.vboxsf -o "$mountOptions" "$name" "$dir" 2>/dev/null; then
```
Comment (tianon) [inline rootfs/rootfs/etc/rc.d/automount-shares]: Why this change too?  `mount -t xyz` is supposed to shell out to `mount.xyz` appropriately as necessary by itself.

Comment (kofj) [inline rootfs/rootfs/etc/rc.d/automount-shares]: @tianon Ok,it is not necessary.Just add `iocharset` is well.

Comment (tianon) [inline rootfs/rootfs/etc/rc.d/automount-shares]: Mind swapping this line back so we can merge? :+1:

**Comment by SvenDowideit** [comment]
@kofj are you going to make this change?

File: rootfs/rootfs/etc/rc.d/automount-shares (hunk 2 above)
Comment (kofj) [inline rootfs/rootfs/etc/rc.d/automount-shares]: yes.you can merge it.

**Comment by kofj** [comment]
@SvenDowideit yes,but just 'iocharset=utf8'.

**Comment by jdiprizio** [comment]
Could we please get this change merged (iocharset)?
